### PR TITLE
added new algorithm shellsort

### PR DESCRIPTION
**Pull request by gardrek**

### CODE REVIEW

File: src/algorithms/shell.rs [hunk 1]
```diff
@@ -0,0 +1,40 @@
+//! [Shell sort](https://en.wikipedia.org/wiki/Shellsort)
+
+use algorithms::Algorithm;
+use array::Array;
+
+/// [Bubble sort](https://en.wikipedia.org/wiki/Bubble_sort)
```
Comment (dmitmel) [inline src/algorithms/shell.rs]: Could you, please, update this link too?

**Review by dmitmel — CHANGES_REQUESTED** [review]

File: src/algorithms/shell.rs (hunk 1 above)
Comment (gardrek) [reply src/algorithms/shell.rs]: I have updated the code, but I am not sure how to add it here

**Review by dmitmel — APPROVED** [review]

**Comment by dmitmel** [comment]
@gardrek Since you are the first person to contribute to this project, I'd like to ask you a few questions and get some feedback:

1. Was the algorithm API easy to use?
2. Was the documentation helpful?
3. Were other algorithm implementations simple?

**Comment by gardrek** [comment]
> 1. Was the algorithm API easy to use?
> 2. Was the documentation helpful?
> 3. Were other algorithm implementations simple?

1. Yes, I'd say so. I didn't use many of the features tho, like the manual coloring.
2. I didn't actually look at any documentation, I just read the code. It wasn't too hard to follow.
3. I only looked at bubble sort and insertion sort closely. As I understood it, they were the closest to Shell sort, so I planned to adapt one or the other into Shell sort. However, in the end I wasn't able to figure it out, and instead just worked from other Shell sort implementations, and my understanding of Shell sort.
